File: src/com/owncloud/android/lib/resources/activities/models/PreviewObject.java
/*  Nextcloud Android Library is available under MIT license
 *   Copyright (C) 2017 Joas Schilling
 *
 *   @author Joas Schilling
 *
 *   Permission is hereby granted, free of charge, to any person obtaining a copy
 *   of this software and associated documentation files (the "Software"), to deal
 *   in the Software without restriction, including without limitation the rights
 *   to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 *   copies of the Software, and to permit persons to whom the Software is
 *   furnished to do so, subject to the following conditions:
 *
 *   The above copyright notice and this permission notice shall be included in
 *   all copies or substantial portions of the Software.
 *
 *   THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
 *   EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
 *   MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
 *   NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS
 *   BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN
 *   ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
 *   CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
 *   THE SOFTWARE.
 *
 */
package com.owncloud.android.lib.resources.activities.models;

/**
 * PreviewObject Data Model
 */

public class PreviewObject {

    private int fileId;
    private String source;
    private String link;
    private Boolean isMimeTypeIcon;
    private String mimeType;
    private String view;

    public PreviewObject() {
    }

    public String getLink() {
        return link;
    }

    public void setLink(String link) {
        this.link = link;
    }

    public String getSource() {
        return source;
    }

    public void setSource(String source) {
        this.source = source;
    }

    public Boolean isMimeTypeIcon() {
        return isMimeTypeIcon;
    }

    public void setMimeTypeIcon(Boolean mimeTypeIcon) {
        isMimeTypeIcon = mimeTypeIcon;
    }

    public String getMimeType() {
        return mimeType;
    }

    public void setMimeType(String mimeType) {
        this.mimeType = mimeType;
    }

    public int getFileId() {
        return fileId;
    }

    public void setFileId(int fileId) {
        this.fileId = fileId;
    }

    public String getView() {
        return view;
    }

    public void setView(String view) {
        this.view = view;
    }
}

File: library/src/main/java/com/owncloud/android/lib/resources/activities/models/PreviewObject.kt
/*
 * Nextcloud Android Library
 *
 * SPDX-FileCopyrightText: 2017-2024 Nextcloud GmbH and Nextcloud contributors
 * SPDX-FileCopyrightText: 2022 Álvaro Brey <alvaro@alvarobrey.com>
 * SPDX-FileCopyrightText: 2017 Joas Schilling <coding@schilljs.com>
 * SPDX-License-Identifier: MIT
 */
package com.owncloud.android.lib.resources.activities.models

/**
 * PreviewObject Data Model
 */
class PreviewObject() {
    var fileId = 0
    var source: String? = null
    var link: String? = null
    var mimeTypeIcon: Boolean? = null
    var mimeType: String? = null
    var view: String? = null
    var filename: String? = null

    @Suppress("LongParameterList")
    constructor(
        fileId: Int,
        source: String?,
        link: String?,
        mimeTypeIcon: Boolean?,
        mimeType: String?,
        view: String?,
        filename: String?
    ) : this() {
        this.fileId = fileId
        this.source = source
        this.link = link
        this.mimeTypeIcon = mimeTypeIcon
        this.mimeType = mimeType
        this.view = view
        this.filename = filename
    }
}
